appendable-rs/appendable/src/handler/jsonl_handler.rs

```diff
@@ -0,0 +1,127 @@
+use crate::index_file::{Index, IndexFile, IndexKey};
+use crate::io::DataHandler;
+use serde_json::{Deserializer, Map, Value};
+use std::fs::File;
+use std::io::{BufRead, BufReader, Read, Seek, SeekFrom};
+use twox_hash::xxh3::hash64;
+
+pub struct JSONLHandler {
+    pub file: File,
+    pub buffer: Option<Vec<u8>>,
+}
+impl JSONLHandler {
+    pub fn new(file: File) -> Self {
+        JSONLHandler { file, buffer: None }
+    }
+
+    /// we need to read the entire file to buffer to compute byte slices for checksums
+    pub fn read_file_to_buffer(&mut self) -> Result<(), String> {
+        let mut buffer = Vec::new();
+        self.file
+            .read_to_end(&mut buffer)
+            .map_err(|e| e.to_string())?;
+
+        self.buffer = Some(buffer);
+
+        Ok(())
+    }
```

Comment:
so instead of reading line by line, we store the entire contents of the file in memory with a buffer.

Originally I wanted to read line by line, but as you see below when we compute checksums, we'll need to compute byte slices

Reply:
It won't be tractible to store the entire file in memory. For the checksums, if we use xxhash then xxhash can be used incrementally so the entire file doesn't have to be checksummed: https://github.com/Cyan4973/xxHash#example

Reply:
Ok. I found the Rust implementation of `xxHash`. That is: https://github.com/DoumanAsh/xxhash-rust?tab=readme-ov-file#example

I'll refactor to use this